### PR TITLE
New Batch API support

### PR DESCRIPTION
**Pull request by annatisch**

### CODE REVIEW

**Comment by annatisch** [comment]
@lmazuel - This is ready to merge if you think everything looks properly set up for release :)

File: azure-batch/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,16 @@
 Release History
 ===============
 
+2.0.0 (2017-02-17)
+++++++++++++++++++
+
+- AAD token authentication now supported.
+- Support for running tasks under a predefined or automatic user account.
```
Comment (jasper-schneider) [inline azure-batch/HISTORY.rst]: Should we mention the changed method names like pool_list_pool_usage_metrics -> pool_list_usage_metrics, or is that too detailed for this history file?

Comment (jasper-schneider) [inline azure-batch/HISTORY.rst]: Also, do we want to clarify that the run as user changes apply to all "task-y" things (task, job manager task, start task, job prep, job release), but the scoped auth only applies to task and job manager task?

File: azure-mgmt-batch/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,17 @@
 Release History
 ===============
 
+3.0.0 (2017-02-17)
+++++++++++++++++++
+
+- Updated `BatchAccount` model - support for pool allocation in the users subscription.
```
Comment (jasper-schneider) [inline azure-mgmt-batch/HISTORY.rst]: users -> user's here since it's possessive?

File: azure-mgmt-batch/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,17 @@
...
+- Updated `BatchAccount` model - support for referencing an Azure Key Vault when allocation pools in users subscription.
```
Comment (jasper-schneider) [inline azure-mgmt-batch/HISTORY.rst]: "allocating pools in the user's subscription" ?
"for accounts created with a pool allocation mode of UserSubscription" is another way I've seen in the Swagger spec.

**Pull request merged**